kernel/signal.c

```diff
@@ -437,3 +491,47 @@ long myst_handle_host_signal(siginfo_t* siginfo, mcontext_t* mcontext)
 {
     return _handle_one_signal(siginfo->si_signo, siginfo, mcontext);
 }
+
+int myst_signal_altstack(const stack_t* ss, stack_t* old_ss)
+{
+    int ret = 0;
+    myst_thread_t* me = myst_thread_self();
+
+    if (old_ss != NULL)
+    {
+        *old_ss = me->signal.altstack;
+        if (old_ss->ss_sp == 0)
+            // Trying to be compatible with Linux. A never-installed
+            // alt stack means it's disabled.
+            old_ss->ss_flags |= SS_DISABLE;
+    }
+
+    if (ss != NULL)
+    {
+        // We don't support swapcontext in libc. No point
+        // to support SS_AUTODISARM which was introduced to enable it.
+        // The only flag allowed here is SS_DISABLE.
+        if (ss->ss_flags & ~(SS_DISABLE) != 0)
+            ERAISE(-EINVAL);
+
+        if (ss->ss_size < MINSIGSTKSZ)
```

Comment:
Should this be checked after handling SS_DISABLE? Acc to man page - 
>  In this case, the kernel ignores any other flags in ss.ss_flags and the remaining fields in ss.

Reply:
Thanks for catching that. Will move.